Add all csv info to displayed table on successful task creation

MichaelBuessemeyer opened the pull request:
This PR unifies the information that is displayed in the table that is shown upon successful task creating with the information of the CSV.

Imo this change does not make so much sense. Maybe I misunderstood something but to me, the display information in the modal is far too much to be easily understood/is not clearly arranged. Additionally, the file name might get too huge if all IDs (like suggested) are added to the file name. Thus I limited this to a maximum of 3 task ids.

I think some general input on my points is needed before we start a code review :)

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- Got to the admin->task view
- Open the task creation
- Create a task and upon success see new content in the modal.
- Then download the CSV and check the file name.

### Issues:
- fixes #5359

------
- [ ] Updated [(unreleased) changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- ~~[ ] Updated [(unreleased) migration guide](../blob/master/MIGRATIONS.unreleased.md#unreleased) if applicable~~
- ~~[ ] Updated [documentation](../blob/master/docs) if applicable~~
- ~~[ ] Adapted [wk-connect](https://github.com/scalableminds/webknossos-connect) if datastore API changes~~
- ~~[ ] Needs datastore update after deployment~~
- [x] Ready for review

MichaelBuessemeyer commented:
@fm3 As you created the issue for this: Do you have an opinion about my thought about whether the changes actually make sense? Or do you have some suggestions to improve these changes?

fm3 commented:
Thanks for looking into this! You are quite right, these changes create somewhat weird behavior. I wrote a message to the original requesting user asking to clarify, let’s see what they reply :) Should have done that immediatly when writing the issue, I’m sorry if this created redundant work!

daniel-wer commented:
The expected changes have been clarified, see https://discuss.webknossos.org/t/show-task-ids-for-created-tasks-miss-filename/1621/6.
Please request a review once you incorporated those :)

fm3 commented:
tbh I’d slightly deviate from the `IDs_`-prefix suggested there, and go with the more general `tasks_ids_<projectnames>_<date>.csv` or something like that.

I also don’t like hiding the short success list, as I still believe it to be useful when creating few tasks. But maybe we can make the CSV download button more prominent even if the short list is there. And maybe the list can be styled differently to more obviously be something different than the CSV. That way, users who want to use the CSV workflow can always do that, and users who like instant feedback on (few) created tasks can still get it. What do you think?

daniel-wer commented:
> tbh I’d slightly deviate from the IDs_-prefix suggested there, and go with the more general tasks_ids_<projectnames>_<date>.csv or something like that.

Sounds fair to me.

> I also don’t like hiding the short success list, as I still believe it to be useful when creating few tasks. But maybe we can make the CSV download button more prominent even if the short list is there. And maybe the list can be styled differently to more obviously be something different than the CSV. That way, users who want to use the CSV workflow can always do that, and users who like instant feedback on (few) created tasks can still get it. What do you think?

It could be hidden behind a "Details" toggle, like this to de-clutter the popup by default?
<img src="https://i.stack.imgur.com/SoCuC.jpg" alt="details-toggle" width="500px">

Also, from the post it sounded like the main issue was that the CSV download button was only visible if a lot of tasks were created. So that should be changed to be always visible.

fm3 commented:
> It could be hidden behind a "Details" toggle, like this to de-clutter the popup by default?

Yes, looks nice :) but that might not even be necessary if the real problem is the accessability of the CSV download button.

> Also, from the post it sounded like the main issue was that the CSV download button was only visible if a lot of tasks were created. So that should be changed to be always visible.

I agree. I think it is always visible already, but should be moved above the list, so the user does not have to scroll to get there.

MichaelBuessemeyer commented:
Thank you two for already discussing the issue :+1:.
I'll try to implement your ideas :)

MichaelBuessemeyer commented:
I just pushed the following changes:
- I updated the csv file name according to @fm3 suggestion and removed the spaces in the date formatting.
- I made the Download CSV button more prominent.
- Added a warning when creating tasks for multiple projects at once

daniel-wer commented on frontend/javascripts/admin/task/task_create_form_view.js:
```diff
@@ -117,14 +117,13 @@ export function downloadTasksAsCSV(tasks: Array<APITask>) {
     return;
   }
-  const maybeTaskPlural = tasks.length > 2 ? "tasks" : "task";
+  const maybeTaskPlural = tasks.length > 2 ? "task_ids" : "task_id";
```
Shouldn't this be

```suggestion
  const maybeTaskPlural = tasks.length > 1 ? "task_ids" : "task_id";
```

MichaelBuessemeyer replied:
correct :facepalm:

daniel-wer reviewed (commented):
Very nice, works well :+1: 

How would I go about testing the "failed" path, so that some of the tasks fail during creation? If I change the dataset name, taskTypeId, or projectName I only end up with error toasts, but not with the popup and "the creation of x tasks failed".

Also, I noticed that when I try to create a task for a project that is not existing, I end up with an unhelpful error toast `Couldn't find the requested resource.` without further details. @fm3 Would it be easy to improve that? Otherwise, I'll create a follow up issue :)

fm3 commented:
@daniel-wer I’ll have a look at the too-generic message, but I’d still track it in a separate issue.

To add a failed task to the report, you can modify line 426 in TaskCreationService.scala to `result <- TaskCreationResult.fromTaskJsFoxes(Fox.failure("Nope") :: taskJsons, warnings)` (prepends a failure to the result list)

daniel-wer reviewed (approved):
Great, works very well :+1:
